android/src/ti/bluetooth/TiBluetoothModule.java

```diff
@@ -8,301 +8,251 @@
+	private BluetoothLeScanner btScanner;
+
+	private int currentState = 0;
+	private boolean isScanning = false;
+
+	@Kroll.constant
+	public static final int MANAGER_STATE_UNKNOWN = 0;
+	@Kroll.constant
+	public static final int MANAGER_STATE_UNSUPPORTED = 1;
+	@Kroll.constant
+	public static final int MANAGER_STATE_UNAUTHORIZED = 2;
+	@Kroll.constant
+	public static final int MANAGER_STATE_POWERED_OFF = 10;
+	@Kroll.constant
+	public static final int MANAGER_STATE_POWERED_ON = 12;
+	@Kroll.constant
+	public static final int MANAGER_STATE_RESETTING = 5;
+
+	@Kroll.constant
+	public static final int SCAN_MODE_BALANCED = ScanSettings.SCAN_MODE_BALANCED;
+	@Kroll.constant
+	public static final int SCAN_MODE_LOW_LATENCY = ScanSettings.SCAN_MODE_LOW_LATENCY;
+	@Kroll.constant
+	public static final int SCAN_MODE_LOW_POWER = ScanSettings.SCAN_MODE_LOW_POWER;
+	@Kroll.constant
+	public static final int SCAN_MODE_OPPORTUNISTIC = ScanSettings.SCAN_MODE_OPPORTUNISTIC;
+
+	@Kroll.constant
+	public static final int PROFILE_ADP = BluetoothProfile.A2DP;
+
+	@Kroll.constant
+	public static final int PROFILE_HEADSET = BluetoothProfile.HEADSET;
+	@Kroll.constant
+	public static final int PROFILE_HEALTH = BluetoothProfile.HEALTH;
+
+	@Kroll.constant
+	public static final int DEVICE_BOND_BONDED = BluetoothDevice.BOND_BONDED;
+	@Kroll.constant
+	public static final int DEVICE_BOND_BONDING = BluetoothDevice.BOND_BONDING;
+	@Kroll.constant
+	public static final int DEVICE_BOND_NONE = BluetoothDevice.BOND_NONE;
+
+	@Kroll.constant
+	public static final int TYPE_L2CAP = BluetoothSocket.TYPE_L2CAP;
+	@Kroll.constant
+	public static final int TYPE_RFCOMM = BluetoothSocket.TYPE_RFCOMM;
+	@Kroll.constant
+	public static final int TYPE_SCO = BluetoothSocket.TYPE_SCO;
+
+	public final int DEFAULT_SCAN_MODE = SCAN_MODE_BALANCED;
+	private int scanmode = DEFAULT_SCAN_MODE;
+
+	public TiBluetoothModule() {
+		super();
+		appContext = TiApplication.getInstance();
+		activity = appContext.getCurrentActivity();
+		appContext.registerReceiver(new BlutoothStateChangedBroadcastReceiver(
+				TiBluetoothModule.this, btAdapter), new IntentFilter(
+				BluetoothAdapter.ACTION_STATE_CHANGED));
+	}
+
+	@Kroll.onAppCreate
+	public static void onAppCreate(TiApplication app) {
+		Log.d(LCAT, "inside onAppCreate");
+	}
+
+	private final ScanCallback scanCallback = new ScanCallback() {
+		@Override
+		public void onScanResult(int callbackType, ScanResult result) {
+			BluetoothDevice device = result.getDevice();
+			if (device != null) {
+				BluetoothDeviceProxy btDeviceProxy = new BluetoothDeviceProxy(
+						device);
+				// TODO resultpayload is array of btDeviceProxy's
+				Log.d(LCAT, "Found something " + device.getName());
+				if (device.getName() != null) {
+					Log.d(LCAT,
+							"Found: " + device.getName() + " "
+									+ device.getAddress());
+
+					ArrayList<String> ids = new ArrayList<String>();
+					for (ParcelUuid id : device.getUuids()) {
```

Comment:
See latest master: https://github.com/hansemannn/ti.bluetooth/blob/master/android/src/ti/bluetooth/TiBluetoothModule.java#L135 and add a null check so it won't crash here